src/wallet/walletActions.js

```diff
@@ -1,7 +1,53 @@
+import steem from 'steem';
 import { createAction } from 'redux-actions';
+import { createAsyncActionType } from '../helpers/stateHelpers';
+import { getTransactionHistory, getDynamicGlobalProperties } from '../helpers/apiHelpers';
 
 export const OPEN_TRANSFER = '@wallet/OPEN_TRANSFER';
 export const CLOSE_TRANSFER = '@wallet/CLOSE_TRANSFER';
+export const GET_GLOBAL_PROPERTIES = createAsyncActionType('@wallet/GET_GLOBAL_PROPERTIES');
+export const GET_USER_TRANSACTIONS = createAsyncActionType('@users/GET_USER_TRANSACTIONS');
+export const GET_USER_EST_ACCOUNT_VALUE = createAsyncActionType(
+  '@users/GET_USER_EST_ACCOUNT_VALUE',
+);
 
 export const openTransfer = createAction(OPEN_TRANSFER);
 export const closeTransfer = createAction(CLOSE_TRANSFER);
+
+export const getGlobalPropertiesSuccess = payload => ({
+  type: GET_GLOBAL_PROPERTIES.SUCCESS,
+  payload,
+});
+
+export const getGlobalProperties = () => dispatch =>
+  getDynamicGlobalProperties().then((result) => {
+    dispatch(getGlobalPropertiesSuccess(result));
+  });
+
+export const getUserTransactionsSuccess = (username, transactions) => ({
+  type: GET_USER_TRANSACTIONS.SUCCESS,
+  payload: {
+    username,
+    transactions,
+  },
+});
+
+export const getUserTransactions = username => (dispatch) => {
```

Comment:
Wouldn't it be better to use `redux-promise-middleware` for this (and in other actions that are working with promises, i.e. `getGlobalProperties`, `getUserEstAccountValue`)?

Reply:
@Sekhmet was kinda confused at using `redux-promise-middleware` at first, but looking into it,  I was able to refactor these actions using it 👍